title_belt_nhl/service/nhl_api.py

```diff
@@ -0,0 +1,44 @@
+import requests
+from title_belt_nhl.static.nhl_tms import nhl_team_abbvs
+from title_belt_nhl.models.nhl_team_schedule_response import ApiTeamScheduleResponse, Game
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from functools import lru_cache
+from typing import Dict, List
+
+
+@lru_cache(maxsize=None)
+def getTeamSchedule(tm_abv:str, season:str) -> ApiTeamScheduleResponse:
+    """Gets the full season schedule for the given team."""
+    url = f'https://api-web.nhle.com/v1/club-schedule-season/{tm_abv}/{season}'
+    response = requests.get(url)
+    if response.status_code == 200:
+        return response.json()
+    else:
+        print(f"Failed to retrieve data. Status code: {response.status_code}")
+    pass
+
+def getFullSchedule(season:str='20242025') -> List[Game] :
+    """
+    Gets the full season schedule.  Have not found an endpoint that will 
+    do this in one call, so we're looping through all teams (in parallel) 
+    to get the full league schedule...
+    """
```

Comment:
I think this is https://api-web.nhle.com/v1/schedule/now?

Reply:
Endpoint only shows given gameweek.  Leaving as is for now.